src/libraries/Common/tests/System/IO/VirtualDriveHelper.Windows.cs

```diff
@@ -0,0 +1,160 @@
+            {
+                // Mount the folder to a drive letter
+                _virtualDriveLetter = CreateVirtualDrive(VirtualDriveTargetDir);
+            }
+            return _virtualDriveLetter;
+        }
+    }
+
+    ///<summary>
+    /// On Windows, mounts a folder to an assigned virtual drive letter using the subst command.
+    /// subst is not available in Windows Nano.
+    /// </summary>
+    public char CreateVirtualDrive(string targetDir)
+    {
+        if (!OperatingSystem.IsWindows())
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        char driveLetter = GetNextAvailableDriveLetter();
+        bool success = RunProcess(CreateProcessStartInfo("cmd", "/c", SubstPath, $"{driveLetter}:", targetDir));
+        if (!success || !DriveInfo.GetDrives().Any(x => x.Name[0] == driveLetter))
+        {
+            throw new InvalidOperationException($"Could not create virtual drive {driveLetter}: with subst");
+        }
+        return driveLetter;
+
+        // Finds the next unused drive letter and returns it.
+        char GetNextAvailableDriveLetter()
+        {
+            List<char> existingDrives = DriveInfo.GetDrives().Select(x => x.Name[0]).ToList();
+
+            // A,B are reserved, C is usually reserved
+            IEnumerable<int> range = Enumerable.Range('D', 'Z' - 'D');
+            IEnumerable<char> castRange = range.Select(x => Convert.ToChar(x));
+            IEnumerable<char> allDrivesLetters = castRange.Except(existingDrives);
+
+            if (!allDrivesLetters.Any())
+            {
+                throw new ArgumentOutOfRangeException("No drive letters available");
+            }
+
+            return allDrivesLetters.First();
+        }
+    }
+
+    /// <summary>
+    /// On Windows, unassigns the specified virtual drive letter from its mounted folder.
+    /// </summary>
+    public void DeleteVirtualDrive(char driveLetter)
+    {
+        if (!OperatingSystem.IsWindows())
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        bool success = RunProcess(CreateProcessStartInfo("cmd", "/c", SubstPath, "/d", $"{driveLetter}:"));
+        if (!success || DriveInfo.GetDrives().Any(x => x.Name[0] == driveLetter))
+        {
+            throw new InvalidOperationException($"Could not delete virtual drive {driveLetter}: with subst");
+        }
+    }
+
+    private static ProcessStartInfo CreateProcessStartInfo(string fileName, params string[] arguments)
+    {
+        var info = new ProcessStartInfo
+        {
+            FileName = fileName,
+            UseShellExecute = false,
+            RedirectStandardOutput = true
+        };
+
+        foreach (var argument in arguments)
+        {
+            info.ArgumentList.Add(argument);
+        }
+
+        return info;
+    }
+
+    private static bool RunProcess(ProcessStartInfo startInfo)
+    {
+        var process = Process.Start(startInfo);
```

Comment:
Process implements `IDisposable`

```suggestion
        using var process = Process.Start(startInfo);
```